enterprise_subsidy/apps/api/v1/views/subsidy.py

```diff
@@ -180,16 +196,33 @@ def base_queryset(self):
         For list actions, some non-strict subset of this is what's returned by ``get_queryset()``.
         """
         kwargs = {}
+        sort_order = self.request.query_params.get("sort_by") or "uuid"
+
         if self.requested_enterprise_customer_uuid:
             kwargs.update({"enterprise_customer_uuid": self.requested_enterprise_customer_uuid})
         if self.requested_subsidy_uuid:
             kwargs.update({"uuid": self.requested_subsidy_uuid})
+        if self.requested_subsidy_uuid_param:
+            kwargs.update({"uuid": self.requested_subsidy_uuid_param})
+        if self.requested_subsidy_title_param:
+            kwargs.update({"title": self.requested_subsidy_title_param})
 
         return Subsidy.objects.filter(**kwargs).prefetch_related(
             # Related objects used for calculating the ledger balance.
             "ledger__transactions",
             "ledger__transactions__reversal",
-        ).order_by("uuid")
+        ).order_by(sort_order)
+
+    def list(self, request, *args, **kwargs):
+        """
+        List subsidies. Adds total number of pages to the response.
+
+        Endpoint Location: GET /api/v1/subsidies/
+        """
+        response = super().list(request, *args, **kwargs)
+        page_size = self.request.query_params.get('page_size') or self.paginator.page_size
+        response.data['page_count'] = ceil(response.data['count'] / int(page_size))
+        return response
```

Comment:
nit: I think you could better encapsulate this part inside the `get_paginated_response()` method of your new pagination class.
https://www.django-rest-framework.org/api-guide/pagination/#example